qdrant-landing/content/articles/qdrant-1.2.x.md

```diff
@@ -0,0 +1,206 @@
+    },
+    {
+      "name": "Osaka",
+      "population": 2.7,
+      "area": 223
+    },
+    {
+      "name": "Kyoto",
+      "population": 1.5,
+      "area": 827.8
+    }
+  ]
+}
+```
+
+We want to filter out the results to include the countries with a city with over 2 million citizens 
+and an area bigger than 500 square kilometers but no more than 1000. There is no such a city in 
+Japan, looking at our data, but if we wrote the following filter, it would be returned:
+
+```json
+{
+  "filter": {
+    "must": [
+      {
+        "key": "country.cities[].population",
+        "range": {
+          "gte": 2
+        }
+      },
+      {
+        "key": "country.cities[].area",
+        "range": {
+          "gt": 500,
+          "lte": 1000
+        }
+      }
+    ]
+  },
+  "limit": 3
+}
+```
+
+Japan would be returned because Tokyo and Osaka match the first criteria, while Kyoto fulfills 
+the second. But that's not what we wanted to achieve. That's the motivation behind introducing 
+a new type of nested filter.
+
+```json
+{
+  "filter": {
+    "must": [
+      {
+        "nested": {
+          "key": "country.cities",
+          "filter": {
+            "must": [
+              {
+                "key": "population",
+                "range": {
+                  "gte": 2
+                }
+              },
+              {
+                "key": "area",
+                "range": {
+                  "gt": 500,
+                  "lte": 1000
+                }
+              }
+            ]
+          }
+        }
+      }
+    ]
+  },
+  "limit": 3
+}
+```
+
+The syntax is consistent with all the other supported filters and enables new possibilities. In 
+our case, it allows us to express the joined condition on a nested structure and make the results 
+list empty but correct.
+
+## Important changes
+
+The latest release focuses not only on the new features but also introduces some changes making 
+Qdrant even more reliable.
+
+### Fixed memory errors on startup
+
+There has been an issue in memory-constrained environments, such as cloud, happening when users were 
+pushing massive amounts of data into the service using `wait=false`. This data influx resulted in an 
+overreaching of disk or RAM limits before the Write-Ahead Logging (WAL) was fully applied. This 
+situation was causing Qdrant to attempt a restart and reapplication of WAL, failing recurrently due 
+to the same memory constraints and pushing the service into a frustrating crash loop with many 
+Out-of-Memory errors.
+
+Qdrant 1.2 automatically detects if memory usage exceeds a specific threshold, informs a user with an 
+error message in the logs, and makes the service halt the loading of collection data and commence 
+operations in a partial state. This state allows for removing collections but doesn't support search 
+or update functions. If Qdrant enters this locked state, a reboot that either increases available or 
+reduces required resources resolves it.
```

Comment:
It is not quite clear to me what this refers to.

I assume this bit refers to recovery mode. But I think this text doesn't really match the way it actually works:

- AFAIK there is no detection with a 'threshold', rather, if Qdrant crashes on startup due to memory errors it can restart in recovery mode if enabled
- recovery mode is disabled by default

More info here: https://github.com/qdrant/docs/blob/v1.2.x/qdrant/v1.2.x/administration.md#recovery-mode

Reply:
Yeah, the original idea was to make this a lot more _automagic_. But detecting out-of-memory situations is extremely hard and error prone. Therefore we changed the approach and made it a lot simpler, as described above.